src/sugar3/activity/activity.py

```diff
@@ -1069,12 +1100,81 @@ def _keep_failed_dialog_response_cb(self, alert, response_id):
             self._session.will_quit(self, False)
 
     def can_close(self):
-        '''
-        Activities should override this function if they want to perform
-        extra checks before actually closing.
-        '''
+        """Activities should override this function if they want to perform
+        extra checks before actually closing."""
 
-        return True
+        if (self._jobject.metadata['title'] != self._jobject_clone_title):
+            self._skip_save_datafiles = False
+            self._pre_naming = True
+            return False
+
+        else:
+
+            if self._is_resumed is True:
+                self._overwrite_alert = OverwriteAlert()
+                self._overwrite_alert.props.title = _(
+                    "The object name already exists.")
+                self._overwrite_alert.props.msg = _(
+                    'Would you like to overwrite current instance or choose a new name?')
+                self.add_alert(self._overwrite_alert)
+                self._overwrite_alert.show()
+                self._overwrite_hid = self._overwrite_alert.connect('response',
+                                                                    self.__overwrite_response_cb)
+            else:
+                self._show_saveas_alert()
+
+            return False
+
+    def __overwrite_response_cb(self, alert, response_id):
+        self.remove_alert(alert)
+
+        # 'Save as' option
+        if response_id == Gtk.ResponseType.OK:
+            self._show_saveas_alert()
+
+        # Overwrite option
+        if response_id == Gtk.ResponseType.CANCEL:
+            self._skip_save_datafiles = False
+            datastore.delete(self._jobject.get_object_id())
```

Comment:
On choosing overwrite the data gets saved onto the original instance whereas the duplicate gets deleted.